### PR TITLE
MacOS Quartz HiDPI temporary fix

### PR DESCRIPTION
**Pull request by LecrisUT**
Please check if this breaks anything in linux quartz. #504 
I've tried to test if this is MacOS specific with `macports` x11/XQuartz, but I couldn't get it to work fullscreen.
@maehne Could you help with this?

### CODE REVIEW

**Comment by maehne** [comment]
Sorry, don't have really the bandwidth the coming two weeks for looking into this due to a very dense teaching schedule.

Also, I don't fully understand what you would like to test: pdfpc with GTK+ rendering through the Quartz API or pdfpc with GTK+ rendering through X11-API (using XQuartz.app server)? I can only easily test pdfpc with GTK+ using the Quartz API. I tested pdfpc with XQuartz some time ago and remember the I observed some issues with the GUI. Due to the less well integration with the rest of the macOS user interface, I went back to using gtk3 +quartz.

**Comment by LecrisUT** [comment]
Wanted to confirm if on x11 MacOS: When you plug in a HiDPI monitor (scaled) and the presenter screen is there, does the screen zoom in when you enable tools (e.g. pointer keyboard shortcut `2`). You could check if the timer disappears outscreen.

I haven't had luck with `xquartz` either. Maybe that option should be removed

File: src/classes/window/fullscreen.vala [hunk 1]
```diff
@@ -182,7 +182,7 @@ namespace pdfpc.Window {
             var monitor_geometry = this.monitor.get_geometry();
             this.window_w = monitor_geometry.width;
             this.window_h = monitor_geometry.height;
-            if (Pdfpc.is_Wayland_backend() && Options.wayland_workaround) {
+            if (Pdfpc.is_Wayland_backend() || Pdfpc.is_Quartz_backend() && Options.wayland_workaround) {
```
Comment (fnevgeny) [inline src/classes/window/fullscreen.vala]: Are you sure this is needed for mac? In any case, this certainly breaks the intended Wayland-related functionality.

Comment (LecrisUT) [reply src/classes/window/fullscreen.vala]: Hmm, I forgot to add parentheses.

Comment (fnevgeny) [reply src/classes/window/fullscreen.vala]: > Hmm, I forgot to add parentheses.

Right, but the main question is whether such a workaround is needed under macOS in the first place. If yes, it should probably be renamed.

Comment (LecrisUT) [reply src/classes/window/fullscreen.vala]: If I remember correctly, when I ran without `-W` it didn't fix the scaling. But let me double check tomorrow. I'll get back to you tomorrow unless the university decides to close abruptly.

File: src/classes/window/fullscreen.vala [hunk 2]
```diff
@@ -182,7 +182,7 @@ namespace pdfpc.Window {
             var monitor_geometry = this.monitor.get_geometry();
             this.window_w = monitor_geometry.width;
             this.window_h = monitor_geometry.height;
-            if (Pdfpc.is_Wayland_backend() || Pdfpc.is_Quartz_backend() && Options.wayland_workaround) {
+            if ((Pdfpc.is_Wayland_backend() || Pdfpc.is_Quartz_backend()) && Options.wayland_workaround) {
```
Comment (AndreasBilke) [inline src/classes/window/fullscreen.vala]: Is `Options.wayland_workaround` a good name then? Maybe `scaling_workaround`

File: src/display_backend.c [hunk 3]
```diff
@@ -54,3 +57,13 @@ bool is_X11_backend() {
...
+bool is_Quartz_backend() {
+    GdkDisplay *gdk_display = gdk_display_get_default();
+    #ifdef GDK_WINDOWING_QUARTZ
+    if (GDK_IS_QUARTZ_DISPLAY(gdk_display)) {
+        return true;
+    }
+    #endif
+
+    return false;
+}
```
Comment (fnevgeny) [inline src/display_backend.c]: Please fix the no-newline here.

Comment (LecrisUT) [reply src/display_backend.c]: You mean delete the newline between `#endif` and `return false;`. Should also change for the functions above?

Comment (fnevgeny) [reply src/display_backend.c]: No, there is no newline at the end of this file.

**Comment by LecrisUT** [comment]
Please do so, sorry for my poor programming etiquette.

**Comment by fnevgeny** [comment]
> Please do so, sorry for my poor programming etiquette.

No problem at all!

**Comment by fnevgeny** [comment]
@AndreasBilke are you fine with the fix as is (or still prefer changing the name of the option)?

**Comment by AndreasBilke** [comment]
For the current patch I think the name of the variable is okay (since it
just controls the behaviour for wayland backends)

**Comment by fnevgeny** [comment]
Fix icon rendering under macOS/Quartz with HiDPI

**Pull request merged**